codegen/import.go

```diff
@@ -29,3 +33,83 @@ func (s *ImportSpec) Code() string {
 	}
 	return fmt.Sprintf(`"%s"`, s.Path)
 }
+
+// getMetaTypeInfo gets type and import info from an attribute's metadata. struct:field:type can have 3 arguments,
+// first being the go type name, second being import path,
+// and third being the name of a qualified import, in case of name collisions.
+func getMetaTypeInfo(att *expr.AttributeExpr) (typeName string, importS *ImportSpec) {
+	if att == nil {
+		return typeName, importS
+	}
+	if args, ok := att.Meta["struct:field:type"]; ok {
+		if len(args) > 0 {
+			typeName = args[0]
+		}
+		if len(args) > 1 {
+			importS = &ImportSpec{Path: args[1]}
+		}
+		if len(args) > 2 {
+			importS.Name = args[2]
+		}
+	}
+	return typeName, importS
+}
+
+// GetMetaTypeImports parses the attribute for all user defined imports
+func GetMetaTypeImports(att *expr.AttributeExpr) []*ImportSpec {
```

Comment:
I believe this function needs to be protected against infinite recursions. Type A may have an attribute of type B which itself has an attribute of type A for example. Take a look at [collectUserTypes](https://github.com/goadesign/goa/blob/v2/http/codegen/service_data.go#L2299) for an example on how to do that. Thank you!